models/migrations/v67.go

```diff
@@ -0,0 +1,141 @@
+// Copyright 2018 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package migrations
+
+import (
+	"strings"
+
+	"code.gitea.io/gitea/models"
+	"code.gitea.io/gitea/modules/log"
+
+	"github.com/go-xorm/xorm"
+)
+
+func reformatAndRemoveIncorrectTopics(x *xorm.Engine) (err error) {
+	log.Info("This migration could take up to minutes, please be patient.")
+	type Topic struct {
+		ID   int64
+		Name string `xorm:"unique"`
+	}
+
+	sess := x.NewSession()
+	defer sess.Close()
+
+	const batchSize = 100
+	var ids []int64
+	touchedRepo := make(map[int64]struct{})
+	topics := make([]*Topic, 0, batchSize)
+
+	if err := sess.Begin(); err != nil {
+		return err
+	}
+	log.Info("Validating existed topics...")
+	for start := 0; ; start += batchSize {
+		topics = topics[:0]
+		if err := x.Asc("id").Limit(batchSize, start).Find(&topics); err != nil {
```

Comment:
If you deleted some topics, this will ignore some topics.

Reply:
Fixed it, Thanks!

Reply:
@lunny Why do you think so? How can I check this wrong behavior?

Reply:
@axifive because you have deleted the topics when your traversing the topics.

Reply:
@lunny, Ok, I'll rewrite this.